### PR TITLE
remove gateway completely

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
remove gateway completely

## Motivation and Context
gateway code has been deprecated for over close
to 2yrs now, with this change we are allowing
for full transition towards server deployments.

## How to test this PR?
Nothing should change 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [x] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: cmd/object-api-datatypes.go [hunk 1]
```diff
@@ -197,7 +192,6 @@ func (o ObjectInfo) ArchiveInfo() []byte {
 		return nil
 	}
 	if len(z) > 0 && z[0] >= 32 {
-		// FS/gateway mode does base64 encoding on roundtrip.
```
Comment (klauspost) [inline cmd/object-api-datatypes.go:200]: This entire block can be removed.

**Review by klauspost — APPROVED** [review]

File: cmd/object-api-datatypes.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/object-api-datatypes.go:200]: done

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by poornas — APPROVED** [review]

**Review by dvaldivia — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#aeddf336-b720-42ea-a98e-eca4ae5b7784) |
<a name="aeddf336-b720-42ea-a98e-eca4ae5b7784"></a>
### 15929-29fd620/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.7:30394
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 0f88c8ee5154:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 2 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 43 seconds
(4/14) Running aws-sdk-ruby tests ... done in 6 seconds
(5/14) Running awscli tests ... done in 1 minutes and 2 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 16 seconds
(8/14) Running minio-go tests ... done in 51 seconds
(9/14) Running minio-java tests ... FAILED in 28 seconds
{
  "name": "minio-java",
  "function": "getObjectRetention()",
  "duration": 45,
  "status": "FAIL",
  "error": "java.time.format.DateTimeParseException: Text '2022-10-27T23:58:58.3Z' could not be parsed at index 19 >>> [io.minio.Xml.unmarshal(Xml.java:55), io.minio.MinioAsyncClient.lambda$getObjectRetention$35(MinioAsyncClient.java:1738), java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:642), java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506), java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2073), io.minio.S3Base$1.onResponse(S3Base.java:559), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(9/14) Running minio-js tests ... done in 50 seconds
(10/14) Running minio-py tests ... done in 1 minutes and 41 seconds
(11/14) Running s3cmd tests ... done in 15 seconds
(12/14) Running s3select tests ... done in 4 seconds
(13/14) Running versioning tests ... done in 3 minutes and 5 seconds

Executed 13 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by dginhoux** [comment]
ok, i understand why my /data stored on nfs share doesn't work since this release.
i'll find a minio alternative now...

**Comment by harshavardhana** [comment]
> ok, i understand why my /data stored on nfs share doesn't work since this release. i'll find a minio alternative now...

👍🏽

**Comment by jbub** [comment]
Is there any migration guide on how to move from filesystem to the new setup ?

**Comment by klauspost** [comment]
@jbub https://github.com/minio/docs/pull/628

**Pull request merged**